Implement new Send modal UI & support token display in confirmed tx rows and tx details modal

nop33 opened the pull request:
I've created a new PR since there are quite many new changes added, let me know if you want me to merge it with the other one.

Closes #72

https://user-images.githubusercontent.com/1579899/220391508-00e7aff9-4576-4503-8a4e-45dedf0980b9.mp4

nop33 commented on .vscode/settings.json:
```diff
@@ -1,4 +1,8 @@
   "typescript.tsdk": "node_modules/typescript/lib",
-  "typescript.preferences.importModuleSpecifier": "non-relative"
+  "typescript.preferences.importModuleSpecifier": "non-relative",
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": true
+  },
+  "eslint.validate": ["javascript", "typescript"]
```
This is magic. No more linting errors, they are just solved on save.

nop33 commented on src/App.tsx:
```diff
@@ -53,12 +53,12 @@ const App = () => {
     s.addresses.status,
     s.activeWallet.isPassphraseUsed,
-    s.tokens
+    s.assetsInfo
```
I decided to rename the slice because `asset` is a more accurate name than `token`, since ALPH is not a token, but a coin. Adding the ALPH info (decimals, name, symbol, etc) in the assets slice simplifies our code a lot (this way we don't have to add special cases to treat ALPH separately from tokens. They are both "assets" :)

mvaivre replied:
Yes, I thought about this as well as I was designing. That's perfect 💯

nop33 commented on src/index.css:
```diff
@@ -44,6 +44,7 @@ button {
   outline: none;
   border: none;
   background: transparent;
+  font-family: 'Inter', sans-serif;
```
For some reason, HTML buttons default to `Arial`...

nop33 commented on src/modals/AddressDetailsModal.tsx:
```diff
@@ -44,7 +44,7 @@ const AddressDetailsModal = ({ addressHash, onClose }: AddressDetailsModalProps)
       onClose={onClose}
       label={t('Address details')}
-      width={670}
+      width={800}
```
I increased the width to give more space for the tokens

mvaivre commented:
Great job! Super sleek. Some things I encountered along the way:

- Transactions are shown twice in the list, even after a full app reload:

<img width="937" alt="image" src="https://user-images.githubusercontent.com/6715943/221427442-50a05042-d9cf-4c80-8165-4ac54f7ea83b.png">

- "Discover addresses" doesn't work. I get the following error `Uncaught (in promise) Error: Could not generate addresses, active wallet not found`

- This one is on me (it was wrong in the mockup) : the amount input in the Send modal shouldn't be in lighter color when not focused

nop33 commented:
> Transactions are shown twice in the list, even after a full app reload:

Fixed, good catch ✅ 

> NIT: We could make sure that an address without label is not displayed with its hash shown twice:

Good recommendation, done ✅ 

> Let's remove the border around the contact name in the input, it's gonna be lighter, less tight.

✅ 

> the amount input in the Send modal shouldn't be in lighter color when not focused

✅ 

> Hashes: let's try to use the inter font in tabular mode instead of the current mono font, to see how it looks

I remember encountering some problems with this here: https://github.com/alephium/desktop-wallet/issues/436

> "Discover addresses" doesn't work. I get the following error Uncaught (in promise) Error: Could not generate addresses, active wallet not found

Thanks for testing this, I hadn't tested it yet. Fixed ✅

nop33 commented:
@mvaivre lemme know if I can merge this :) I look forward to do that so I can merge master into v2.0 to get the latest updates we pushed to the 1.5.0 release.

mvaivre commented:
All good! Thanks for the fixes! Merging.

DeafEyeJedi commented:
Thanks @Ilias for your never ending energy on this one... looking forward to the updates!

> On Mar 1, 2023, at 9:17 AM, Mikaël Vaivre ***@***.***> wrote:
> 
> ﻿
> All good! Thanks for the fixes! Merging.
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.
